src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,966 @@
+The `getReader()` method of the `ReadableStream` interface creates a reader and locks the stream to
+it. While the stream is locked, no other reader can be acquired until this one is released.
+
+The [`read()`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader/read)
+method of the `ReadableStreamDefaultReader` interface returns a promise providing access to the next
+chunk in the stream's internal queue. It fulfills or rejects with a result depending on the state of
+the stream. The different possibilities are as follows:
+
+- If a chunk is available, the promise will be fulfilled with an object of the form
+  `{ value: chunk, done: false }`.
+- If the stream becomes closed, the promise will be fulfilled with an object of the form
+  `{ value: undefined, done: true }`.
+- If the stream becomes errored, the promise will be rejected with the relevant error.
+
+#### The `locked` property
+
+You can check if a readable stream is locked by accessing its
+[`ReadableStream.locked`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/locked)
+property.
+
+### Readable stream code samples
+
+The code sample below shows all the steps in action. I first create a `ReadableStream` that in its
+`underlyingSource` argument defines a `start` method that tells the stream's `controller` to
+`enqueue()` the current time during 10 seconds and then to `close()` the stream. I consume this
+stream by creating a reader via the `getReader()` method and calling `read()` until the stream is
+`done`.
+
+```js
+let interval = null;
+
+const stream = new ReadableStream({
+  start(controller) {
+    interval = setInterval(() => {
+      // Extract just the hh:mm:ss from the date.
+      let string = new Date().toISOString().replace(/.*?T(.*?)\..*?$/, '$1 ');
+      // Add the string to the stream.
+      console.log(`Enqueued ${string}`);
+      controller.enqueue(string);
+    }, 1_000);
+
+    setTimeout(() => {
+      clearInterval(interval);
+      // Close the stream after 10s.
+      controller.close();
+    }, 10_000);
+  },
+
+  cancel() {
+    // This is called if the reader cancels.
+    clearInterval(interval);
+  },
+});
+
+let result = '';
+const reader = stream.getReader();
+
+const readStream = async () => {
+  // The `read()` method returns a promise that
+  // resolves when a value has been received.
+  const { done, value } = await reader.read();
+  // Result objects contain two properties:
+  // `done`  - `true` if the stream has already given you all its data.
+  // `value` - Some data. Always `undefined` when `done` is `true`.
+  if (done) {
+    return console.log(`Stream complete.\n${result}`);
+  }
+  result += value;
+  console.log(`Read ${result.length} characters so far`);
+  console.log(`Most recently read chunk: ${value}`);
+  // Read some more, and call this function again.
+  return readStream();
+};
+readStream();
```

Comment:
I think a recursive function is harder to understand than a loop. Eg:

```suggestion
async function concatStringStream(stream) {
  let result = '';
  const reader = stream.getReader();
  while (true) {
    // The `read()` method returns a promise that
    // resolves when a value has been received.
    const { done, value } = await reader.read();
    // Result objects contain two properties:
    // `done`  - `true` if the stream has already given you all its data.
    // `value` - Some data. Always `undefined` when `done` is `true`.
    if (done) return result;
    result += value;
    console.log(`Read ${result.length} characters so far`);
    console.log(`Most recently read chunk: ${value}`);
  }
}

concatStringStream(stream)
  .then((result) => console.log('Stream complete', result));
```

Reply:
Went for your suggestion.